src/main.rs

```diff
@@ -49,7 +49,13 @@ use alacritty::util::fmt::Red;
 fn main() {
     // Load command line options and config
     let options = cli::Options::load();
-    let config = load_config(&options);
+    let mut config = load_config(&options);
+
+    // Override configurations based on CLI options.
+    if options.title.is_some() {
+        let mut dynamic_title = config.dynamic_title_mut();
+        *dynamic_title = false;
+    }
```

Comment:
As mentioned in the comment about the `config.rs` change, this could be changed to just

```rust
if options.title.is_some() {
    config.disable_dynamic_title();
}
```